govcd/vm.go

```diff
@@ -1352,26 +1353,16 @@ func (vm *VM) UpdateInternalDisks(disksSettingToUpdate *types.VmSpecSection) (*t
 		return nil, fmt.Errorf("cannot update internal disks - VM HREF is unset")
 	}
 
-	task, err := vm.UpdateInternalDisksAsync(disksSettingToUpdate)
+	description := vm.VM.Description
+	vm, err := vm.UpdateVmSpecSection(disksSettingToUpdate, description)
 	if err != nil {
 		return nil, err
 	}
-	err = task.WaitTaskCompletion()
-	if err != nil {
-		return nil, fmt.Errorf("error waiting for task completion after internal disks update for VM %s: %s", vm.VM.Name, err)
-	}
-	err = vm.Refresh()
-	if err != nil {
-		return nil, fmt.Errorf("error refreshing VM %s: %s", vm.VM.Name, err)
-	}
+
 	return vm.VM.VmSpecSection, nil
 }
 
-// UpdateInternalDisksAsync applies disks configuration for the VM.
-// types.VmSpecSection has to have all internal disk state. Disks which don't match provided ones in types.VmSpecSection
-// will be deleted. Matched internal disk will be updated. New internal disk description found
-// in types.VmSpecSection will be created.
-// Returns Task and error.
+// Deprecated, use UpdateVmSpecSectionAsync instead
```

Comment:
This deprecation is not mentioned in the changelog.
Would it be more precise to suggest using either `UpdateInternalDisks` (which is not deprecated) or `UpdateVmSpecSectionAsync`

Reply:
Added to changelog, updated the comment aswell.